modular/gater/object_handler.go

```diff
@@ -203,14 +203,14 @@ func (g *GateModular) getObjectHandler(w http.ResponseWriter, r *http.Request) {
 		log.CtxErrorw(reqCtx.Context(), "failed to download object", "error", err)
 		return
 	}
-	w.Write(data)
 	w.Header().Set(model.ContentTypeHeader, objectInfo.GetContentType())
 	if isRange {
 		w.Header().Set(model.ContentRangeHeader, "bytes "+util.Uint64ToString(uint64(low))+
 			"-"+util.Uint64ToString(uint64(high)))
 	} else {
 		w.Header().Set(model.ContentLengthHeader, util.Uint64ToString(objectInfo.GetPayloadSize()))
 	}
+	w.Write(data)
```

Comment:
why change the location for write body?

Reply:
@will-2012 told me that frontend peers found if the write of data is before the header, it will cause the headers to not be written correctly. This was fixed in develop branch by Will, but not in this refactor code, so adding it here.